applications/file-split-ftp/src/main/java/org/springframework/integration/samples/filesplit/Application.java

```diff
@@ -0,0 +1,229 @@
+import org.springframework.integration.dsl.mail.Mail;
+import org.springframework.integration.file.FileHeaders;
+import org.springframework.integration.file.remote.session.SessionFactory;
+import org.springframework.integration.file.splitter.FileSplitter;
+import org.springframework.integration.file.support.FileExistsMode;
+import org.springframework.integration.ftp.session.DefaultFtpSessionFactory;
+import org.springframework.integration.http.config.EnableIntegrationGraphController;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.MessagingException;
+import org.springframework.web.servlet.config.annotation.CorsRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
+
+@SpringBootApplication
+@EnableIntegrationGraphController
+public class Application {
+
+	private static final String EMAIL_SUCCESS_SUFFIX = "emailSuccessSuffix";
+
+	public static void main(String[] args) {
+		SpringApplication.run(Application.class, args);
+	}
+
+	@Autowired
+	private MailProperties mailProperties;
+
+	/**
+	 * Poll for files, add an error channel, split into lines route the start/end markers
+	 * to {@link #markers()} and the lines to {@link #lines()}.
+	 * @return the flow.
+	 */
+	@Bean
+	public IntegrationFlow fromFile() {
+		return IntegrationFlows.from(Files.inboundAdapter(new File("/tmp/in"))
+					.preventDuplicates(false)
+					.patternFilter("*.txt"), e -> e.poller(Pollers.fixedDelay(5000)))
+				.enrichHeaders(h -> h.header(MessageHeaders.ERROR_CHANNEL, "tfrErrors.input"))
+				.handle(Files.splitter(true, true))
+				.<Object, Class<?>>route(Object::getClass, m -> m
+						.channelMapping(FileSplitter.FileMarker.class, "markers.input")
+						.channelMapping(String.class, "lines.input"))
+				.get();
+	}
+
+	/**
+	 * Process lines; append (no flush) to the appropriate file.
+	 * @return the flow.
+	 */
+	@Bean
+	public IntegrationFlow lines() {
+		return f -> f.enrichHeaders(h -> h.headerExpression(
+						FileHeaders.FILENAME, "payload.substring(1, 4) + '.txt'", true))
+				.handle(Files.outboundAdapter("'/tmp/out'")
+						.appendNewLine(true)
+						.fileExistsMode(FileExistsMode.APPEND_NO_FLUSH), // files remain open for efficiency
+							e -> e.id("fileOut"));
+	}
+
+	/**
+	 * Process file markers; ignore START, when END, flush the files, ftp them and
+	 * send an email.
+	 * @return the flow.
+	 */
+	@Bean
+	public IntegrationFlow markers() {
+		return f -> f.<FileSplitter.FileMarker>filter(m -> m.getMark().equals(FileSplitter.FileMarker.Mark.END),
+						e -> e.id("markerFilter"))
+				.publishSubscribeChannel(s -> s
+
+						// first trigger file flushes
+						.subscribe(sf -> sf.transform("'/tmp/out/.*\\.txt'", e -> e.id("toTriggerPattern"))
+										.handle("fileOut.handler", "trigger", e -> e.id("flusher")))
```

Comment:
Having the problem with bean definitions order on Linux we should move `Files.outboundAdapter()` to the separate `@Bean`. At least until the fix for https://github.com/spring-projects/spring-integration-java-dsl/issues/119